### DISCUSSION TITLE
2.9.5 Coral Module Downloads Broken

### DISCUSSION THREAD

**Discussion by grid8** [codeproject/CodeProject.AI-Server#241] [General]
Running 2.9.5 on Linux with Coral TPU. Working with the MobileNet module but you cannot download any other module. Always results in the below.

Unable to download and install objectdetection-yolov8-medium-edgetpu.zip: Unable to download 'objectdetection-yolov8-medium-edgetpu.zip'. Error: Could not find a part of the path '/usr/bin/codeproject.ai-server-2.9.5/downloads/modules/packages/ObjectDetectionCoral/objectdetection-yolov8-medium-edgetpu.zip'.

**Comment by claptons** [comment]
need create Folder "ObjectDetectionCoral"  on /usr/bin/codeproject.ai-server-2.9.5/downloads/modules/packages/, finish path is 
/usr/bin/codeproject.ai-server-2.9.5/downloads/modules/packages/ObjectDetectionCoral/. Then try download  YOLOv8 model

**Reply by grid8** [reply]
That did the trick, appreciate it.

**Comment by Sailing74** [comment]
Thanks for the tip, claptons. It just helped me on a new install as well.
Can we open this discussion again - as the installer apparently forgets to create this path which is required?

**Reply by claptons** [reply]
No problem, yes you can

**Comment by ChrisMaunder** [comment]
I've reopened this.

Were you running the server as sudo?

I think a better solution here is to have the downloads NOT be under /usr/bin. Maybe something as obvious as ~/Downloads.

**Reply by Sailing74** [reply]
I ran all install steps - and also launched the server - with sudo/as root, yes. (I guess this also happens if people use the provided systemd config for starting the server without editing the user.)

I wouldn't expect ~./Downloads to exist by default. I may be an outlier, but I'm running things on a VM with the server variant installed. And that variant does not create the fancy stuff in the user's home folder. (And I guess no variant does ist for /root.) 

Wouldn't /var be a good place for downloads? (/tmp might not have the required size. I believe that was a problem for previous versions.)
